python/tk_multi_loader/dialog.py

```diff
@@ -926,6 +926,20 @@ def _history_navigate_to_item(self, preset, item):
         finally:
             self._history_navigation_mode = False
 
+    def _get_item_from_entity(self, ctx, model):
+        """
+        Retrieve the item object based on
+        entity type and entity id.
+
+        :param Sgtk Context ctx: Context object.
+        :param model: The SG model.
+
+        :returns: Model item object or None if not found.
+        """
+        if ctx.task:
+            return model.item_from_entity(ctx.task["type"], ctx.task["id"])
+        return model.item_from_entity(ctx.entity["type"], ctx.entity["id"])
```

Comment:
Maybe a less verbose approach:

```suggestion
        ctx_object = ctx.task or ctx.entity
        return model.item_from_entity(ctx_object["type"], ctx_object["id"])
```

Reply:
In this case there will always be a ctx.entity in the context. So, what I'm looking for is that in case there is a Task related to the current context, ctx.task takes precedence over ctx.entity. But I get your point, maybe a ternary operator here would be less verbose.